In [1]:
import math
import scipy as sp
from scipy import special

# 1.9 Excercises

## Counting

### 1. How many ways are there to permutate the letters in the word MISSISSIPPI?

**R/** First thing we must do is know how many letters are there in the word MISSISSIPPI.

In [2]:
w = "MISSISSIPPI"
len(w)

11

So now we now there are $11!$ ways to permutate all the letters.

But we only care about distinguishable permutations, an $11!$ is overcounting by the factor of permutations of the repetitions of each character. To solve this problem we need to count the total number or repetitions for each letter in the word:

$$ M = 1$$
$$ I = 4$$
$$ S = 4$$
$$ P = 2$$

And then divide $11!$ by those factors. So the final answer is:

$$ \dfrac{11!}{1!\cdot 4!\cdot 4!\cdot 2!} = \dfrac{11!}{2(4!)^{2}} $$

Which we can then calculate:

In [3]:
math.factorial(11) / (2*(math.factorial(4)**2))

34650

### 2. (a) How many 7-digit phone numbers are possible, assuming that the first digit can't be a 0 or a 1?

**R/** This can easily be solved via the multiplication rule:
For the first spot we can only pick eight numbers (all but 0 or 1), for the six remaning spots we can pick 10 diferent numbers each.

$$ 8\cdot 10^{6} $$

And we can calculate it:


In [4]:
8 * 10**6

8000000

An other way to think about this problem is to calculate all the posible 7-digit phone numbers:

$$ 10^{7} $$

Then, we need to substract all the numbers which start with 0s or 1s. We can pick 2 numbers for the first spot (0s or 1s), and then pick 10 numbers for each of the remaining six spots:

$$ 10^{7} - (2 \cdot 10^{6}) $$

We can calculate this solution too: 

In [5]:
(10**7) - (2 * 10**6)

8000000

### 2. (b) Resolve (a), except now assume also that the pone number is not allowed to start with 911.

**R/** We can use as a starting point our solution to (a). Now we need to substract all possible permutations where number starts with 911:

For the first spot there's one number we can pick (9), for the other two spots too (1). So the posible permutations which start with 911 are:

$$ 1 \cdot 1 \cdot 1 \cdot 10^{4} $$

So our answear is:

$$ (8 \cdot 10^{6}) - 10^{4} $$

That is:

In [6]:
(8 * 10**6) - 10**4

7990000

### 3. Fred is planning to go out to dinner each night of a certrain week (Monday through Friday), with each dinner being at one of his ten favorite restaurants.

#### a) How many possibilities are there for Fred's schedule if Fred is not willing to eat at the same restaurant more than once?

**R/** This is immediate from the multiplication rule. Fred can pick from
* 10 restaurants on Monday.
* 9 on Tuesday.
* 8 on Wenesday.
* 7 on Thursday
* 6 on Friday

That is $ 10 \cdot 9 \cdot 8 \cdot 7 \cdot 6 $ posible schedules for Fred

A fancy (and useful for larger problems) way to write this is:

$$ \dfrac{10!}{(10-5)!} $$

We can calculate this:

In [7]:
math.factorial(10) / math.factorial(10-5)

30240

#### b) How many possibilities are there for Fred's schedule if Fred is willing to eat at the same restaurant more than once, but is not willing to eat at the same place twice in a row?


**R/** This sounds like a more complicated problem. Fred can pick from

* 10 options available on Monday
* 9 options available on Tuesday (since he can't eat at the same place twice in a row)

Here comes the tricky part:
* 9 options available on Wenesday (since he can't pick the restaurant from Tuesday, but can pick the one from Monday)
* 9 options available on Thursday (since he can't pick the restaurant from Wenesday, but can pick the one from Monday or Tuesday)
* 9 options available on Friday (since he can't pick the restaurant from Thursday, but can pick the one from Monday, Tuesday or Wenesday)

The answear then is: 

$$ 10 \cdot 9^{4} $$

In [8]:
10 * 9**4

65610

### 4. A *round-robin tournament* is being held with n tennis players; this means every player will play against every other player exactly once.

#### a) How many possible outcomes are there for the tournament (the outcome list out who won and who lost for each tournament)

**R/** This is the most difficult problem so far. It's difficulty lays in the fact that we have to combine two ways of counting to solve this problem:

1. We need to count all posible pairs we can make with n players. 
2. We need to count all posible outcomes for each pair.

To count all posible games:
$$ \binom{n}{2} = \dfrac{n!}{2! \cdot (n-2)!} $$

 
(We can use binomial coefficient because a player can't play with itself and it doesn't matter which player we choose first, it's the same game).

We know each game can only have two outcomes, that means there are $2^{g}$ posible outcomes given $g$ games. But we already solved how many games are there, so our answear:

$$ 2^{\binom{n}{2}} $$ 

#### b) How many games are played in total?

I believe Joseph laid this question after the first only to give you a hint if you couldn't solve a) first. We already have our answear:

$$ \binom{n}{2} = \dfrac{n!}{2! \cdot (n-2)!} $$

### 5. A *knock-out tournament*  is being held with $2^{n}$ players


#### a) How many rounds are there?

**R/** $n$.

This is inmediate from the anatomy of the tournament. If we have $2^{n}$ players and know the last round is where we have two players left, we just reverse engineer our number of players because $n$ es the factor by which we multiplied our players.  

#### b) Count how many games in total are played, by adding up the numbers of games played in each round.

To solve this we need:
1. First find out how many rounds are there
2. Count how many games are there per round
3. Sum them up

We already know there are $n$ rounds.

We know there are $\dfrac{2^{n}}{2^{n-(n-1)}}$ games in the first round. 
There are $\dfrac{2^{n}}{2^{n-(n-2)}} $ games in the second round and so on for it.

We can simplify this term by adding:

$$ 2^{n-1} + 2^{n-2}  ...  2^{n-n} $$

#### c) Count how many games are played in total, this time by directly thinking about it without doing almost any calculation.

**R/** With the hint (how many players need to be disqualified?) we inmediately see that there must be $2^{n} - 1$ teams disqualified.

This is importart because then we realizied that one game = one team disqualified. That means that the number of games are:

$$ 2^{n}-1 $$

### 6. There are 20 people at a chess club. They each find opponents and start playing. How many possibilities are there for how they are matched up, assuming that in each game it does matter who has the white pieces.

**R/** The solution to this problem is simple: 

$$ \dfrac{20!}{(20-2)!} $$

We can simplify it (since everything else cancels):

$$ 20 \cdot 19 $$ 

And our answear is:

In [9]:
20*19

380

### 7. To chess players, A and B, are going to play 7 games. Each game has three possible outcomes: a win for A (which is a loss for B), a draw (tie), and a loss for A (which is a win for B). A win is worth 1 point, a draw is worth 0.5 points and a lose is worth 0 points

#### a) How many possible outcomes for the individual games are there, such that overall player A ends up with 3 wins, 2 draws and 2 losses?

**R/** We can imagine each game as a spot in a 7 digit string. Each sport can take any of three letters (w for win, d for draw and l for losses).

Now the problem reduces to how many distinguishable permutations of 7 digits can we make with 3 Ws, 2 Ds and 2 Ls. It is easy to see that this is an isomorphic problem to the Mississippi one. 

So we have $7!$ possible combinations with our string. But we overcounted for:

$$ W = 3 $$
$$ D = 2 $$
$$ L = 2 $$

So we need to divide by those factors:

$$ \dfrac{7!}{3! \cdot 2! \cdot 2!} = \dfrac{7!}{4 \cdot (3!)}$$

So our answear is:

In [10]:
math.factorial(7) / (4 *math.factorial(3))

210

What we just did is the same as $\binom{7}{3}$ for the wins, $\binom{4}{2}$ for the losses and $\binom{2}{2}$ for the draws.


$$ \binom{7}{3} \cdot \binom{4}{2} \cdot \binom{2}{2}$$

#### b) How many possible outcomes for the individual games are there, such that A ends with 4 points and B ends up with 3 points.

**R/** A ending with 4 points and B ending with 3 points can be achivied by different ways:

* A winning 4 games and lossing 3.
* A winning 3 games and drawing 2 games.
* A winning 2 games and drawing 4 games.
* A winning 1 game and drawing 6 games.

If we imagine a 7 digit string again, we now have repeated letters:


* First scenario:  $ A = 4, B = 3 $
* Second scenario:  $ A = 3, D = 2, B = 2 $
* Third scenario:  $ A = 2, D = 4$
* Fourth scenario:  $ D = 6 $





So, if que count all posible permutations we have:

$$ \dfrac{7!}{4! \cdot 3!} +  \dfrac{7!}{3! \cdot 2! \cdot 2!}  + \dfrac{7!}{2! \cdot 4!} + \dfrac{7!}{6!} $$ 

So our answear is:

In [11]:
one = math.factorial(7) / (math.factorial(4) * math.factorial(3))
two = math.factorial(7) / (math.factorial(3) * math.factorial(2) * math.factorial(2))
three = math.factorial(7) / (math.factorial(2) * math.factorial(4))
four = math.factorial(7) / math.factorial(6)

one + two + three + four

357

#### c) Now assume that they are playing a best-of-7 match, where the match will end as soon as either player has 4 points. How many possible outcomes for the individual games are there, such that the match lasts for 7 games and A wins by a score of 4 to 3.

**R/** Best way I can think of this problem is interpreting the condition. For a match to last 7 games and A winning with 4 points there are just two possibilities:

* A has 3 points in firsts 6 games and wins the last game.
* A has 3.5 points in firsts 6 games and ties the last game.

Once we have this, we need to solve two problems: 

A getting 3 points out of six games. (This is the same as we just did before)

* First scenario: A=3, D=0, B=3
* Second scenario: A=2, D=2, B=2
* Third scenario: A=1, D=4, B=1
* Fourth scenario: D=6



$$ \dfrac{6!}{3! \cdot 3!} + \dfrac{6!}{2! \cdot 2! \cdot 2!} + \dfrac{6!}{4!} + \dfrac{6!}{6!} $$


A getting 3.5 points out of six games. (Same thing here)
* First scenario: A=3, D=1 B=2
* Second scenario: A=2, D=3, B=1
* Third scenario: A=1, D=5, B=0


$$ \dfrac{6!}{3! \cdot 2!} + \dfrac{6!}{2! \cdot 3! \cdot 2!} + \dfrac{6!}{5!} $$

So our final answear is:


$$ \left(\dfrac{6!}{3! \cdot 3!} + \dfrac{6!}{2! \cdot 2! \cdot 2!} + \dfrac{6!}{4!} + \dfrac{6!}{6!}\right) + \left(\dfrac{6!}{3! \cdot 2!} + \dfrac{6!}{2! \cdot 3! \cdot 2!} + \dfrac{6!}{5!}\right)$$

Which is:

In [12]:
one = math.factorial(6) / (math.factorial(3) * math.factorial(3))
two = math.factorial(6) / (math.factorial(2) * math.factorial(2) * math.factorial(2))
three = math.factorial(6) / math.factorial(4) 
four = math.factorial(6) / math.factorial(6) 
five = math.factorial(6) / (math.factorial(3) * math.factorial(2))
six = math.factorial(6) / (math.factorial(2) * math.factorial(3))
seven = math.factorial(6) /  math.factorial(5)
                             
one + two + three + four + five + six + seven

267

### 8. a) How many ways are there to split a dozen people into 3 teams, where one team has 2 people and the other teams have 5 people each?

**R/** This problem can be solved with the binomial coefficient. We pick 2 out of 12 people. Then we have 10 people left, from which we need to pick five. The remaining 5 are in the other team. It's important to note that we are overcounting by the factor of two: since it doesn't matter which group of five we pick first. (Picking a group of five is the same as not picking that group).

So our solution is:

$$ \dfrac{\binom{12}{2} \cdot \binom{10}{5}} {2} $$

And we can calculate it:

In [13]:
(sp.special.binom(12, 2) * sp.special.binom(10, 5)) / 2

8316.0

### 8. b) How many ways are there to split a dozen people into 3 teams, where every team has 4 people?

**R/** We pick 4 people out of 12 for the first team, 4 people out of 8 for the second and the remaining people are the third team. In this ocation we are overcounting by the fact of $3!$ since there are $3!$ ways of ways of picking the same three teams in different orders. 

$$ \dfrac{\binom{12}{4} \cdot \binom{8}{4}}{3!} $$

Specific answear is:

In [14]:
(sp.special.binom(12, 4) * sp.special.binom(8, 4)) / (3*2)

5775.0

### 9. a) How many paths are there from point (0,0) to the point (110, 111) in the plane such that each step either consists of going one unit up or one unit to the right?

**R/** We can imagine each movement as right(R) or left(L). There are 110 + 111 movements available, 110 must be right and the remaining 111 must be up.

$$ \binom{221}{110} $$

Our answear:

In [17]:
sp.special.binom(221, 110)

1.8026173520842921e+65

### 9. b) How many paths are there from point (0,0) to the point (210, 211) in the plane such that each step either consists of going one unit up or one unit to the right and the path has to go through (110, 111)?

**R/** We already know there are $\binom{221}{110}$ ways to get from (0,0) to (110,111). Now we just need to multiply it by all the ways to get from (110,111) to (210, 211):

If (110,111) is our starting point, there must be 100 movements up and 100 movements right to get to (210, 211). So 
there are 100 + 100 movements available. We then pick 100 right movemments and the left are up movements:

$$ \binom{200}{100} $$

So our answear is:

$$ \binom{221}{110} \cdot \binom{200}{100} $$

Important note: This time we are not overcounting  since order doesn't matter. Every path is different. 

### 10. To fulfill the requirements for a certrain degree, a student can choose to take any 7 out of a list of 20 courses, with the constraint that at leas 1 of the 7 courses must be a statistics course. Suppose that 5 out of the 20 courses are statistics courses.

#### a) How many choices are there for which 7 courses to take?

**R/** The student has $\binom{20}{7}$ choices. Out of which $\binom{15}{7}$ does not include statistics courses. So our answear is: 

$$\binom{20}{7}-\binom{15}{7} $$

Our answear is:

In [24]:
sp.special.binom(20, 7) - sp.special.binom(15, 7)

71085.0

#### b) Explain intuitively why the answer to (a) is not $\binom{5}{1} \cdot \binom{19}{6}$

**R/** That solution is overcounting. If a student chooses first one statistic course (x) and then chooses an other one for the other six (y) is the same as the student choosing first y and then x.

### 11. Let A and B be sets with |A| = n, |B| = m.

#### a) How many functions are there from A to B. (i.e, functions with domain A assinging an element of B to each element of A).

### 12. Four players, named A, B, C and D, are playing a card game. A standard, well-shuffled, deck of cards is dealt to the players (so each player receives a 13-card hand).

#### a) How many possibilities are there for the hand that player A will get? (Within a hand, the order in which cards were received doesn't matter)

**R/** The posible hands for player A are the same that B, C or D. We just pick 13 cards out of 52.

$$ \binom{52}{13} $$

#### b) How many possibilities are there overall for what hands everyone will get, assuming that it matters which players gets which hand, but not the order of cards within a hand.

**R/** Hand of A player can have 


$$ \binom{52}{13} \cdot \binom{39}{13} \cdot \binom{26}{13} \cdot \binom{13}{13} $$

In [26]:
sp.special.binom(52,13) * sp.special.binom(39,13) * sp.special.binom(26,13)

5.3644737765488799e+28